homeassistant/components/ai_speaker/sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Support for AI-Speaker sensors."""
+from datetime import timedelta
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(minutes=60)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Perform the setup for AI-Speaker status sensor."""
+    _LOGGER.debug("AI-Speaker sensor, async_setup_entry")
+    ais_gate_instance = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([AisSensor(ais_gate_instance)], True)
+
+
+class AisSensor(Entity):
+    """AiSpeakerSensor representation."""
+
+    def __init__(self, ais_gate_instance):
+        """Sensor initialization."""
+        self._ais_gate = ais_gate_instance
+        self._ais_id = None
+        self._ais_product = None
+        self._ais_manufacturer = None
+        self._ais_model = None
+        self._ais_os_version = None
+        self._ais_api_level = None
+        self._ais_ip_address = None
+        self._ais_hostname = None
+
+    @property
+    def device_info(self):
+        """Device info."""
+        return {
+            "identifiers": {(DOMAIN, self._ais_id)},
+            "name": f"AI-Speaker {self._ais_product}",
+            "manufacturer": self._ais_manufacturer,
+            "model": self._ais_model,
+            "sw_version": self._ais_os_version,
+            "via_device": None,
+        }
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, friendly identifier for this entity."""
+        return self._ais_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"AI-Speaker {self._ais_product} status"
+
+    @property
+    def state(self):
+        """Return the status of the sensor."""
+        return self._ais_api_level
```

Comment:
This sensor makes no sense. It's not something that the user will ever act on.

Reply:
I'm going to add as the sensor value the speaker Internet connection spead. This information should be helpfull for user. The api level will go to attributes - hope this will be OK for you?
The real functionality of this integration will be delivered in next pull requests - all the explanation in the comment above.